liquid-rust-typeck/src/checker.rs

```diff
@@ -459,12 +461,74 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
             ir::BinOp::Sub => self.check_arith_op(cursor, BinOp::Sub, ty1, ty2),
             ir::BinOp::Mul => self.check_arith_op(cursor, BinOp::Mul, ty1, ty2),
             ir::BinOp::Div => self.check_arith_op(cursor, BinOp::Div, ty1, ty2),
+            ir::BinOp::Mod => self.check_mod(cursor, ty1, ty2),
             ir::BinOp::Gt => self.check_cmp_op(BinOp::Gt, ty1, ty2),
             ir::BinOp::Lt => self.check_cmp_op(BinOp::Lt, ty1, ty2),
             ir::BinOp::Le => self.check_cmp_op(BinOp::Le, ty1, ty2),
+            ir::BinOp::BitAnd => self.check_bitwise_op(BinOp::And, ty1, ty2),
+        }
+    }
+
+    fn check_bitwise_op(&self, op: BinOp, ty1: Ty, ty2: Ty) -> Ty {
+        match (ty1.kind(), ty2.kind()) {
+            (TyKind::Refine(BaseTy::Bool, e1), TyKind::Refine(BaseTy::Bool, e2)) => TyKind::Refine(
+                BaseTy::Bool,
+                ExprKind::BinaryOp(op, e1.clone(), e2.clone()).intern(),
+            )
+            .intern(),
+            _ => unreachable!(
```

Comment:
I assume you can also have bitwise operation between integers? In that case, this is a `todo!` not an `unreachable!`. But since we are at it, I think it's not too hard to implement it for integers as well. We can just return `i32{true}` in that case for now.

Reply:
Done.